tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go

```diff
@@ -81,3 +89,251 @@ func VerifyOperatorDeployment(apiClient *clients.Settings,
+
+	if crbSa, err := rbac.PullClusterRoleBinding(
+		apiClient,
+		rbacName); err == nil {
+		glog.V(100).Infof("ClusterRoleBinding %q found. Deleting...", rbacName)
+
+		err = wait.PollUntilContextTimeout(
+			context.TODO(),
+			time.Second*15,
+			time.Minute,
+			true,
+			func(ctx context.Context) (bool, error) {
+				err = crbSa.Delete()
+
+				if err != nil {
+					glog.V(100).Infof("Error deleting ClusterRoleBinding %q : %v", rbacName, err)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Deleted ClusterRoleBinding %q", rbacName)
+
+				return true, nil
+			})
+
+		if err != nil {
+			return fmt.Errorf("failed to delete Cluster RBAC %q", rbacName)
+		}
+	}
+
+	return nil
+}
+
+// DeleteServiceAccount deletes the service account and verifies it was removed.
+func DeleteServiceAccount(apiClient *clients.Settings, saName, nsName string) error {
+	glog.V(100).Infof("Removing Service Account")
+	glog.V(100).Infof("Assert SA %q exists in %q namespace", saName, nsName)
+
+	if deploySa, err := serviceaccount.Pull(
+		apiClient, saName, nsName); err == nil {
+		glog.V(100).Infof("ServiceAccount %q found in %q namespace", saName, nsName)
+		glog.V(100).Infof("Deleting ServiceAccount %q in %q namespace", saName, nsName)
+
+		err = wait.PollUntilContextTimeout(
+			context.TODO(),
+			time.Second*15,
+			time.Minute,
+			true,
+			func(ctx context.Context) (bool, error) {
+				err := deploySa.Delete()
+
+				if err != nil {
+					glog.V(100).Infof("Error deleting ServiceAccount %q in %q namespace: %v",
+						saName, nsName, err)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Deleted ServiceAccount %q in %q namespace", saName, nsName)
+
+				return true, nil
+			})
+
+		if err != nil {
+			return fmt.Errorf("failed to delete ServiceAccount %q from %q ns", saName, nsName)
+		}
+	} else {
+		glog.V(100).Infof("ServiceAccount %q not found in %q namespace", saName, nsName)
+	}
+
+	return nil
+}
+
+// DeleteDeployment deletes the deployment and verifies it and all related pods were removed.
+func DeleteDeployment(
+	apiClient *clients.Settings,
+	deploymentName, nsName string) error {
+	glog.V(100).Infof("Removing test deployment %q from %q ns", deploymentName, nsName)
+
+	if deploymentObj, err := deployment.Pull(apiClient, deploymentName, nsName); err == nil {
+		glog.V(100).Infof("Deleting deployment %q from %q namespace", deploymentName, nsName)
+
+		err = deploymentObj.DeleteAndWait(300 * time.Second)
+
+		if err != nil {
+			glog.V(100).Infof("Error deleting deployment %q from %q namespace: %v",
+				deploymentName, nsName, err)
+
+			return fmt.Errorf("failed to delete deployment %q from %q namespace: %w",
+				deploymentName, nsName, err)
+		}
+	} else {
+		glog.V(100).Infof("deployment %q not found in %q namespace", deploymentName, nsName)
+	}
+
+	return nil
+}
+
+// InsureAllPodsRemoved insure all deployment pods in namespace with the specific pod label were removed.
+func InsureAllPodsRemoved(
```

Comment:
shouldn't this be `EnsureAllPodsRemoved` ?

Reply:
yep, fixing, thanks!